File: nekoshlink/shlink-auth-server/src/main/java/org/nekosoft/shlink/oauth2server/UserAdminConfig.java
package org.nekosoft.shlink.oauth2server;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.annotation.Order;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.web.SecurityFilterChain;


@Configuration
public class UserAdminConfig {

    @Bean
    @Order(1)
    public SecurityFilterChain filterChainBasicAdminApiLogin(HttpSecurity http) throws Exception {
        return http
                .requestMatchers().mvcMatchers(
                        "/adm",
                        "/adm/**"
                )
                .and()
                .cors()
                .and()
                .csrf().disable()
                .authorizeRequests(authorize ->
                        authorize.anyRequest().authenticated()
                )
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                .and()
                .httpBasic()
                .and().build();
    }

}

File: nekoshlink/shlink-auth-server/src/main/kotlin/org/nekosoft/shlink/oauth2server/UserAdminConfig.kt
package org.nekosoft.shlink.oauth2server

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.annotation.Order
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.config.web.servlet.invoke
import org.springframework.security.web.SecurityFilterChain


@Configuration
class UserAdminConfig {

    @Bean
    @Order(1)
    fun filterChainBasicAdminApiLogin(http: HttpSecurity): SecurityFilterChain {
        http {
            securityMatcher(
                "/adm",
                "/adm/**",
            )
            cors {  }
            csrf { disable() }
            authorizeRequests {
                authorize(anyRequest, authenticated)
            }
            sessionManagement { sessionCreationPolicy = SessionCreationPolicy.STATELESS }
            httpBasic {  }
        }
        return http.build()
    }

}
